### DISCUSSION TITLE
Choose the download-directory

### DISCUSSION THREAD

**Discussion by hco** [webinstall/webi-installers#289] [Ideas]
Thank you for the great tool! I stumbled upon it while looking for a way to install caddy within a project directory.

I think it would be really valuable to be able to define the directory where packages are installed during the download.

The usecase I am thinking about is having the appropriate tooling installed within a project directory. So let's say your project is located in $HOME/projects/fancy-web-project and it requires NodeJS and Postgres. You could then have a script that installs NodeJS and Postgres into that local directory without interfering with the globaly installed packages.

As a vision we could even think of a `webi.json` similar to how package.jsons work. And then we could do a `curl -sS https://webinstall.dev/webi.json | bash`, where we parse the webi.json and install all listed packages.

In that case the directory should not be added to the PATH, and at some point one should definitively think about having only symlinks in the project directory and the actual installations in $HOME, but I guess that's not necessary for a proof of concept.

**Comment by coolaj86** [comment]
I didn't realize there were discussions, so sorry I didn't see and respond to this earlier.

I like the ideas that you have here.

In fact, the first iteration of webi has a package-scoped installer for an app I was working on at work. I started building it in Go, but abandoned it for the bash approach - but it's waaaaay more difficult to manipulate paths in bash, especially when targeting both BSD (macOS) and Linux.

I think it would actually be easier to handle the symlinks to a project directory rather than trying to download and install to different directories.

Let's keep thinking on it.

If you have some ideas for the implementation, please share.

**Comment by pl643** [comment]
I came across webi by needing to do something similar: installing static binaries to /tmp/.xdg.$USER and adding that folder to PATH for temporary usage.  Once done, I can easily clean up with rm -rf /tmp/.xdg.$USER and not pollute the system with unneeded software/files.

I too would like to see the ability to change the default location of ~/.local.  One way to do this is export a variable like WEBI_INSTALL_DIR=/tmp/webi and webi would use this as the location instead of ~/.local.

pl
